### PR TITLE
ifspeed RA: try to detect interface name from ip address

### PR DESCRIPTION
**Pull request by tomerazran**
We want to be able to detect interface name automatically from IP address in order to be able to use this RA to detect bond interface failures.
Please refer to the thread I started on pacemaker mailing list:
http://lists.clusterlabs.org/pipermail/users/2017-August/006224.html

### CODE REVIEW

File: extra/resources/ifspeed [hunk 1]
```diff
@@ -165,13 +182,40 @@ monitor() {
...
+                # findif couldn't find the interface
+                if ocf_is_probe; then
+                    ocf_log info "[$FINDIF] failed"
+                    exit $OCF_NOT_RUNNING
+                else
+                    ocf_log err "[$FINDIF] failed"
+                    exit $rc
+                fi
+            fi
+        fi
```
Comment (kgaillot) [inline extra/resources/ifspeed]: validate() should just check that the configuration is sane, i.e. that either iface or ip is specified. The rest can go into update()

One benefit is that stops won't fail if the IP is gone

File: extra/resources/ifspeed [hunk 2]
```diff
@@ -7,12 +7,16 @@
...
+#     Add "ip" parameter to detect network interface name by ip address:
+#     http://lists.clusterlabs.org/pipermail/users/2017-August/006224.html
+#
 # OCF instance parameters:
 #    OCF_RESKEY_name:         name of attribute to set in CIB
 #    OCF_RESKEY_iface:        network interface to monitor
 #    OCF_RESKEY_bridge_ports: if not null and OCF_RESKEY_iface is a bridge, list of
 #                             bridge ports to consider.
-#                             Default is all ports which have designated_bridge=root_id 
+#                             Default is all ports which have designated_bridge=root_id
```
Comment (kgaillot) [inline extra/resources/ifspeed]: This is fine, but for future reference, we discourage formatting-only changes (even when it's an improvement). It distracts from the real changes when reviewing, and there's always a chance of an unintentional mistake.

**Comment by jnpkrn** [comment]
I commend using `findif.sh` and not reinventing the wheel, but
dependency-wise, I think the proper pacemaker agents should only
use a limited set of `ocf_*` functions (something that can be viewed
as "OCF shell scripting API", distributed with resource-agents)
and other out-of-band coupling (coincidentally, with files from
the same project) should be strictly discouraged.

In other words, resource-agents suite might be better fit for this
agent when it starts relying on `findif.sh`.

Otherwise, it will be hard to untangle the dependencies once we,
for instance, decide to separate "OCF shell scripting API" as
a separate project on its own.

**Comment by kgaillot** [comment]
I think keeping ifspeed in pacemaker and relying on findif.sh in resource-agents is less than ideal, but acceptable for practical purposes. There isn't as clear a line between pacemaker agents and resource-agents agents as there should be, but the general idea is that agents that only set pacemaker node attributes should be in pacemaker. It would be fine to ask resource-agents to "bless" findif.sh as part of the official interface.

In the longer term, resource agents are overdue for a major overhaul, from the OCF standard to the packaging. But that's not happening anytime soon.

File: extra/resources/ifspeed (hunk 2 above)
Comment (tomerazran) [inline extra/resources/ifspeed]: You are right - I will keep that in mind in the future.

**Comment by kgaillot** [comment]
Looks good. There's been a minor spelling correction to ifspeed since you branched, so please rebase your changes on top of the current master (or merge), and squash your two commits into one. If you update your branch, this pull request will automatically update. (If any of that is unfamiliar, I can give more details.)

**Comment by tomerazran** [comment]
Done.

**Comment by bubble75** [comment]
Last commit seems to be broken...
Also, may I ask you to follow formatting style of the original code please? I mean subshells syntax and spacing. Thank you!

**Comment by kgaillot** [comment]
> Last commit seems to be broken...

Travis is broken currently, it uses an Ubuntu environment for its tests, and it's having some problem setting up the environment. I hope to look at it soon, but it may just be a transient issue with the upstream repositories -- if not, they may be using a newer Ubuntu version that requires some tweaking of our travis config.

**Comment by bubble75** [comment]
I see, but commit is broken by itself - it is not rebased, and contains already merged code.

**Comment by kgaillot** [comment]
@tomerazran Probably the easiest way at this point will be to copy your ifspeed here into a clean checkout. Use the 1.1 branch instead of the master branch, if you can get it done in the next week, because we are close to releasing 1.1.18.

**Comment by tomerazran** [comment]
I checked out from 1.1 and committed my changes there. Is that OK now?

**Comment by tomerazran** [comment]
I got confuse while trying to commit the change.
I'm not sure the push is OK. Can you please verify?

**Comment by bubble75** [comment]
Not OK.
Please rebase on top of 1.1 branch instead of merge.

**Comment by tomerazran** [comment]
I don't know how to. Can you please help me with that?

**Comment by kgaillot** [comment]
@tomerazran Sure -- there's more than one way to do it, but since we just want one commit changing one file here, the easiest will be:
1. Copy your ifspeed somewhere temporarily.
2. Check out the 1.1 branch with "git checkout 1.1"
3. Make sure your branch is up-to-date. I like to add an "upstream" alias with "git remote add upstream https://github.com/ClusterLabs/pacemaker.git", so updating is simply "git pull upstream 1.1".
4. Move your saved ifspeed back into its usual place, and commit it.
5. I'd close this PR and submit a new one, since you want to submit your 1.1 branch, not your master branch.

In more complicated situations (multiple commits to multiple files), it's better to use "git rebase -i", but that can be more involved.

**Comment by tomerazran** [comment]
I tried to create another PR #1361 
Please close this one.

**Comment by kgaillot** [comment]
Thanks

**Pull request closed without merging**